amxmodx/amx.cpp

```diff
@@ -4173,3 +4173,43 @@ int AMXAPI amx_GetStringOld(char *dest,const cell *source,int use_wchar)
   dest[len]='\0';     /* store terminator */
   return AMX_ERR_NONE;
 }
+
+
+#include <chrono>
+#include <amxmodx.h>
+#include <CPlugin.h>
+
+int AMXAPI amx_ExecPerf(AMX* amx, cell* retval, int index)
+{
+    CPluginMngr::CPlugin* perf_Plug = g_plugins.findPluginFast(amx);
+    if (amxmodx_perflog->value > 0.0f && perf_Plug && (perf_Plug->isDebug() || amxmodx_debug->value == 2.0f))
+    {
+        char perf_funcname[sNAMEMAX + 1];
+        perf_funcname[0] = '\0';
+        amx_GetPublic(perf_Plug->getAMX(), index, perf_funcname);
+        if (perf_funcname[0] == '\0')
+            sprintf(perf_funcname, "Unknown_ID%d", index);
+
+        const char* perf_plugname = perf_Plug->getName();
+        if (!perf_plugname || perf_plugname[0] == '\0')
+            perf_plugname = "Unknown_plugin";
+
+        using std::chrono::high_resolution_clock;
+        using std::chrono::duration_cast;
+        using std::chrono::duration;
+        using std::chrono::microseconds;
+
+        auto t1 = high_resolution_clock::now();
+        int err = amx_Exec(amx, retval, index);
+        auto t2 = high_resolution_clock::now();
+
+        auto ms_int = duration_cast<microseconds>(t2 - t1);
+        auto ms_float = (float)(ms_int.count() / 1000.0f);
```

Comment:
Why dont you use `static` variables here ? amx_Exec will be called many a times. 

`t2` and `ms_int` variables are only being used once, may be we can skip its mem allocation and do the following,
```c++
 auto ms_float = (float)(( duration_cast<microseconds>( high_resolution_clock::now() - t1) ).count() / 1000.0f);
```

Reply:
Why you think what static faster than local, this is not PAWN language, but okay I can do it.

Reply:
Again, dont do it just because of my suggestion. If you think local vars are better, just state so, no issues.

Reply:
As I remember `amx_Exec` may be executed recursively, so you can't use `static` variables here.

Reply:
In this case I can't use static variables even if this faster

Reply:
Yes, it can be recursive, calling one function from another.